pkg/k8shandler/collection.go

```diff
@@ -3,31 +3,56 @@ package k8shandler
 import (
 	"fmt"
 	"github.com/openshift/cluster-logging-operator/pkg/utils"
+	"github.com/sirupsen/logrus"
 	"k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
+	"reflect"
 
 	logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1alpha1"
 	sdk "github.com/operator-framework/operator-sdk/pkg/sdk"
+	apps "k8s.io/api/apps/v1"
 )
 
 func CreateOrUpdateCollection(logging *logging.ClusterLogging) (err error) {
-	if err = createOrUpdateFluentdServiceAccount(logging); err != nil {
-		return
-	}
 
-	if err = createOrUpdateCollectionPriorityClass(logging); err != nil {
-		return
-	}
+	if logging.Spec.Collection.Type == "fluentd" {
+		if err = createOrUpdateFluentdServiceAccount(logging); err != nil {
+			return
```

Comment:
Don't we need to return `err` here? Can we continute to modify anything else?

Reply:
no need since we're using named returns!
We want to return out since something failed, it will then retry after the interval time has passed (5 sec)